File: bestia-client/src/Bnet/Message/Inventory/DropItemCMSG.cs
using Bnet;

namespace BestiaBehemothClient.Bnet.Message.Inventory
{
  public partial class DropItemCMSG : ICMSG
  {
    public ulong ItemId { get; set; }
    public uint Amount { get; set; }

    /// Which copy to drop; 0 for a plain stack. See the field comment in drop_item_cmsg.proto.
    public ulong UniqueId { get; set; }

    public override Envelope ToEnvelope()
    {
      var dropItem = new global::Bnet.DropItemCMSG
      {
        ItemId = ItemId,
        Amount = Amount,
        UniqueId = UniqueId
      };

      return new Envelope
      {
        DropItem = dropItem
      };
    }
  }
}

File: zone-server/src/main/kotlin/net/bestia/zone/item/DropItemCMSG.kt
package net.bestia.zone.item

import net.bestia.zone.message.CMSG

data class DropItemCMSG(
    override val playerId: Long,
    val itemId: Long,
    val amount: Int,
    /** Which copy to drop; 0 leaves the choice to the server. See `drop_item_cmsg.proto`. */
    val uniqueId: Long = 0L
) : CMSG {
    companion object {
        fun fromBnet(playerId: Long, bnet: net.bestia.bnet.proto.DropItemCMSGProto.DropItemCMSG): DropItemCMSG {
            return DropItemCMSG(playerId, bnet.itemId, bnet.amount, bnet.uniqueId)
        }
    }
}
